relays/client-substrate/src/chain.rs

```diff
@@ -58,7 +63,20 @@ pub trait Chain: ChainBase + Clone {
 	///
 	/// The chain may suport multiple tokens, but this particular type is for token that is used
 	/// to pay for transaction dispatch, to reward different relayers (headers, messages), etc.
-	type Balance: Parameter + Member + DeserializeOwned + Clone + Copy + CheckedSub + PartialOrd + SaturatingAdd + Zero;
+	type Balance: AtLeast32BitUnsigned
+		+ FixedPointOperand
+		+ Parameter
+		+ Parameter
+		+ Member
+		+ DeserializeOwned
+		+ Clone
+		+ Copy
+		+ Bounded
+		+ CheckedSub
+		+ PartialOrd
+		+ SaturatingAdd
+		+ Zero
+		+ std::convert::TryFrom<sp_core::U256>;
```

Comment:
This `TryFrom` is required, because `payment_queryFeeDetails` returns `InclusionFee<NumberOrHex>` and `NumberOrHex` holds `u64`/`U256`. Not sure what happens on chain with e.g. `U512`

Reply:
I think in other parts of substrate we already require that the balance is at most `128bits`.

Reply:
I haven't seen that, but it is good to know